src/wp-login.php

```diff
@@ -1154,6 +1155,44 @@ function wp_login_viewport_meta() {
 		login_footer();
 		break;
 
+	case 'confirmemail':
+		if ( ! isset( $_GET['id'], $_GET['hash'] ) ) {
+			wp_die( __( 'Missing or invalid key.' ) );
+		}
+
+		if ( ! is_user_logged_in() ) {
+			wp_safe_redirect( wp_login_url() );
+			exit;
+		}
+
+		$user_id   = wp_unslash( $_GET['id'] );
+		$email_key = wp_unslash( $_GET['hash'] );
+
+		if ( ! current_user_can( 'edit_user', $user_id ) ) {
+			wp_die( __( 'Missing or invalid key.' ) );
+		}
+
+		$updated = send_user_email_change_confirmation_process( $user_id, $email_key );
+		if ( ! $updated ) {
+			wp_die( __( 'Missing or invalid key.' ) );
+		}
+
+		/**
+		 * Fires an action hook when the account email has been confirmed by the user.
+		 *
+		 * @since x.x
+		 *
+		 * @param int $user_id User ID.
+		 */
+		do_action( 'user_email_confirmed', $user_id );
```

Comment:
This can be used for things like, redirecting to a custom page after the user confirms their email change - for example, to a custom WordPress (or bbPress) page.